source/main/index.ts

```diff
@@ -317,18 +374,93 @@ const onAppReady = async () => {
 
     await safeExit();
   });
+
+  // If first-instance launched by clicking on a `web+cardano:` URL on Windows/Linux:
+  // (First element is `execPath`, we want to exclude arguments starting with '-' or
+  // '/' (some `/nix/store` paths on Ubuntu.)
+  const plausibleArgs = process.argv.slice(1).filter((arg) => !(arg.startsWith('-') || arg.startsWith('/')));
+  if (plausibleArgs.length > 0) {
+    const lastArg = plausibleArgs[plausibleArgs.length - 1];
+    logger.info(
+      '[Custom-Protocol] will handleCustomProtocol via initial first-instance argv',
+      {
+        url: lastArg,
+        argv: process.argv,
+        plausibleArgs
+      }
+    );
+    handleCustomProtocol(lastArg, mainWindow);
+  }
+
+  // If first-instance launched by `open-url` on Darwin:
+  if (darwinURLWeAreLaunchedWith) {
+    logger.info(
+      '[Custom-Protocol] will handleCustomProtocol via initial open-url',
+      {
+        url: darwinURLWeAreLaunchedWith,
+      }
+    );
+    handleCustomProtocol(darwinURLWeAreLaunchedWith, mainWindow);
+  }
 };
 
+// The following works only on macOS (either notifying already-running or first launch).
+// The handler for `open-url` needs to be set up in `will-finish-launching`, or else, it
+// won’t catch the first URL, if the app was started by clicking on a URL.
+app.on('will-finish-launching', () => {
+  app.on('open-url', (event, url) => {
+    event.preventDefault();
+    if (!mainWindow) {
+      // The app was launched by 'open-url', we're currently just before `app.on('ready')`
+      // we have to handle this URL only after `mainWindow` exists:
+      logger.info('[Custom-Protocol] app launched by open-url', {
+        url,
+      });
+      darwinURLWeAreLaunchedWith = url;
+    } else {
+      // Subsequent 'open-url':
+      logger.info('[Custom-Protocol] will handleCustomProtocol via subsequent open-url', {
+        url,
+      });
+      if (mainWindow.isMinimized()) mainWindow.restore();
+      mainWindow.focus();
+      handleCustomProtocol(url, mainWindow);
+    }
+  });
+});
+
 // Make sure this is the only Daedalus instance running per cluster before doing anything else
-const isSingleInstance = app.requestSingleInstanceLock();
+const isFirstInstance = app.requestSingleInstanceLock();
 
-if (!isSingleInstance) {
-  app.quit();
+if (!isFirstInstance) {
+  safeExitWithCode(0);
```

Comment:
I’m really not sure why this change was introduced – if `safeWithExitCode` for second instance will just `app.exit(0)` anyway… (and had to be modified to do just that, or else there are errors about `log.transports.file.stream` being undefined.